lib/mozart_fetcher/fetcher.ex

```diff
@@ -11,20 +11,35 @@ defmodule MozartFetcher.Fetcher do
 
   def process(components) do
     ExMetrics.timeframe "function.timing.fetcher.process" do
-      max_timeout =
-        TimeoutParser.max(components) +
-          Application.get_env(:mozart_fetcher, :additional_task_await_timeout)
+      max_timeout = TimeoutParser.max(components)
+      stream_opts = [timeout: max_timeout, on_timeout: :kill_task, max_concurrency: 40]
 
       components
       |> Enum.with_index()
-      |> Enum.map(&Task.async(fn -> Component.fetch(&1) end))
-      |> Enum.map(&Task.await(&1, max_timeout))
-      |> decorate_response
+      |> Task.async_stream(&Component.fetch/1, stream_opts)
+      |> Enum.to_list()
+      |> extract_successful()
+      |> decorate_response()
       |> Jason.encode!()
     end
   end
 
   defp decorate_response(envelopes) do
     %{components: envelopes}
   end
+
+  # returns the successful one (including 500 etc) and logs
+  # the tasks which have not completed.
+  # At the moment it's in form  of `[exit: :timeout]` so hard to
+  # easily log which component has failed.
+  defp extract_successful(responses) do
+    {oks, errors} = Enum.split_with(responses, fn {k, _v} -> k == :ok end)
+
+    errors |> Enum.each(&log(&1))
+    oks |> Enum.map(fn {:ok, result} -> result end)
+  end
+
+  defp log({type, error}) do
```

Comment:
Can we have a test for this part.
Does `[exit: :timeout]` match the matcher `{type, error}`? One is a tuple and the other a list. It might be my knowledge about this isn't great, so it might do!

This should be `{exit: reason}` according to the docs and the comment further up. So type would be "exit" and reason would be error here? Maybe it's just naming that makes it a bit confusing.